### PR TITLE
DOCSP-30748: WriteConcern api streamline

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-30748>
Staging:
- [Write concern guide](https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-30748-wc-api-streamline/fundamentals/crud/write-read-pref/#write-concern)
- [transactions pg example](https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-30748-wc-api-streamline/fundamentals/transactions/#example)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/fundamentals/crud/write-read-pref.txt [hunk 1]
```diff
@@ -49,9 +49,9 @@ Options
 
 The {+driver-long+} provides the ``writeconcern`` package, which lets
 you specify the write concern for a replica set. Set the write
-concern using the ``SetWriteConcern()`` method with an ``Option``
-type. The ``Option`` type has the following methods to specify the
-write concern:
+concern using the ``SetWriteConcern()`` method with an ``WriteConcern``
+type. The ``WriteConcern`` type has the following convenience methods to
```
Comment (caitlindavey) [inline source/fundamentals/crud/write-read-pref.txt:53]: Are "convenience methods" a term?

**Review by caitlindavey — APPROVED** [review]

File: source/fundamentals/crud/write-read-pref.txt (hunk 1 above)
Comment (rustagir) [reply source/fundamentals/crud/write-read-pref.txt]: omitted the word convenience

**Review by blink1073 — APPROVED** [review]
LGTM!

File: source/fundamentals/crud/write-read-pref.txt [hunk 2]
```diff
@@ -60,56 +60,110 @@ write concern:
...
-   :emphasize-lines: 2-3
+   :emphasize-lines: 2-6,10-11
    
    uri := "mongodb://<hostname>:<port>"
-   wc := writeconcern.W(2)
-   opts := options.Client().ApplyURI(uri).SetWriteConcern(writeconcern.New(wc))
-   client, err := mongo.Connect(context.TODO(), opts)
+   cliWC := &writeconcern.WriteConcern{
+       W: 2,
+       Journal: false,
```
Comment (matthewdale) [inline source/fundamentals/crud/write-read-pref.txt:158]: `Journal` is a `*bool`, so you can't assign a bool literal to it. It needs to be updated to something like:
```go
journal := false
cliWC := &writeconcern.WriteConcern{
    W: 2,
    Journal: &journal,
}
```

**Pull request merged**